src/subset_matrix.js

```diff
@@ -0,0 +1,115 @@
+import * as scran from "scran.js"; 
+import * as utils from "./utils/general.js";
+
+/**
+ * This step subsets datasets based on a list of indices to continue with the rest of the analysis
+ *
+ * Methods not documented here are not part of the stable API and should not be used by applications.
+ * @hideconstructor
+ */
+export class SubsetCellsState {
+    #qc
+    #parameters;
+    #cache;
+
+    constructor(qc, parameters = null, cache = null) {
+        if (!(qc instanceof qc_module.QualityControlState)) {
+            throw new Error("'qc' should be a State object from './quality_control.js'");
+        }
+        this.#qc = qc;
+
+        this.#parameters = (parameters === null ? {} : parameters);
+        this.#cache = (cache === null ? {} : cache);
+        this.changed = false;
+    }
+
+    free() {
+        utils.freeCache(this.#cache.matrix);
+    }
+
+    /***************************
+     ******** Getters **********
+     ***************************/
+
+    fetchSubsetMatrix() {
+        if (!("matrix" in this.#cache)) {
+            this.#raw_compute(this.#parameters.indices);
+        }
+        return this.#cache.matrix;
+    }
+
+    fetchIndices() {
+        return this.#parameters.indices ? this.#parameters.indices : [];
+    }
+
```

Comment:
Missing a `fetchAnnotations()` function, see the equivalent in the inputs and QC steps.

Why do we even need an exported `fetchIndices()` here anyway? What other step needs to use the indices?